rflx/model.py

```diff
@@ -1877,16 +1894,24 @@ def merged(self) -> "UnprovenMessage":
                     )
                 elif link.source == field:
                     for final_link in inner_message.incoming(FINAL):
-                        structure.append(
-                            Link(
-                                final_link.source,
-                                link.target,
-                                And(link.condition, final_link.condition).simplified(),
-                                link.length,
-                                link.first,
-                                link.location,
-                            )
+                        merged_condition = And(link.condition, final_link.condition)
+                        proof = merged_condition.check(
+                            [
+                                *inner_message.type_constraints(merged_condition),
+                                inner_message.field_condition(final_link.source),
+                            ]
                         )
+                        if proof.result != ProofResult.unsat:
```

Comment:
Is there a reason for not checking `proof.result == ProofResult.sat`?

Reply:
The result could also be `ProofResult.unknown`. The only situation where you know for sure that a field is unreachable is when you have a proof that the path condition is statically false (i.e. `ProofResult.unsat`). For the unknown case, you better don't remove the edge (it could be a prover limitation or whatever, but it's not granted that the condition is false).